serviceerror/resourceExhausted.go

```diff
@@ -34,15 +34,17 @@ type (
 	// ResourceExhausted represents resource exhausted error.
 	ResourceExhausted struct {
 		Cause   enumspb.ResourceExhaustedCause
+		Scope   enumspb.ResourceScope
 		Message string
 		st      *status.Status
 	}
 )
 
 // NewResourceExhausted returns new ResourceExhausted error.
-func NewResourceExhausted(cause enumspb.ResourceExhaustedCause, message string) error {
+func NewResourceExhausted(cause enumspb.ResourceExhaustedCause, scope enumspb.ResourceScope, message string) error {
```

Comment:
While we may think this is only used by us, this is public API at https://pkg.go.dev/go.temporal.io/api/serviceerror and we shouldn't break compatibility IMO. Can we either have a `NewResourceExhaustedWithOptions` or request that those building this with a scope create the struct directly?

Reply:
Okay! I thought only server and go-sdk use this function. Let me make that change.

Reply:
I have removed this argument from this function. It is better to create this struct directly.

Reply:
Thanks! Will approve when api repo PR merged and regen occurs and is merged into this branch

Reply:
Thanks @cretz! I have merge api changes..